get.go

```diff
@@ -181,15 +181,18 @@ func (j *JWKS) refresh() (err error) {
 	if j.givenKeys != nil {
 		for kid, key := range j.givenKeys {
 			// Only overwrite the key if configured to do so.
+			kidIdx := findKeyIdx("", kid, j.keys)
 			if !j.givenKIDOverride {
-				if _, ok := j.keys[kid]; ok {
+				if kidIdx >= 0 {
 					continue
 				}
 			}
-
-			j.keys[kid] = parsedJWK{public: key.inter}
+			if kidIdx >= 0 {
+				j.keys[kidIdx] = ParsedJWK{Public: key.inter, kid: kid}
+			} else {
+				j.keys = append(j.keys, ParsedJWK{Public: key.inter, kid: kid})
```

Comment:
```suggestion
				j.keys[kidIdx] = append(j.keys[kidIdx], ParsedJWK{Public: key.inter, kid: kid})
```
Easy to do `map[string][]key` ...

Reply:
makes sense, changed to map